src/components/TextInput/TextInput.tsx

```diff
@@ -81,29 +93,6 @@ export interface InputProps
   /** Provides text for a `Label` component if `showLabel` is set to true;
    * populates an `aria-label` attribute if `showLabel` is set to false. */
   labelText: string | JSX.Element;
-  /** The max number for a `number` TextInput type. */
-  max?: number;
-  /** The max length of the input field. This prop is for all input types
-   * except for the `number` type. */
-  maxLength?: number;
-  /** The min number for a `number` TextInput type. */
-  min?: number;
-  /** Used to reference the input element in forms. */
-  name?: string;
-  /** The action to perform on the `input`/`textarea`'s onChange function  */
-  onChange?: (
-    event:
-      | React.ChangeEvent<HTMLInputElement>
-      | React.ChangeEvent<HTMLTextAreaElement>
-  ) => void;
-  /** The action to perform on the `input`/`textarea`'s onClick function  */
-  onClick?: (event: React.MouseEvent<HTMLInputElement, MouseEvent>) => void;
-  /** The action to perform on the `input`/`textarea`'s onFocus function  */
-  onFocus?: (event: React.FocusEvent<HTMLInputElement>) => void;
-  /** Regex to query the user input against. */
-  pattern?: string;
-  /** Populates the placeholder for the input/textarea elements */
-  placeholder?: string;
```

Comment:
I clearly remember when we were building this component and the steps we had to go through to make it truly functional by adding all these props. :)

Reply:
Feels nice to be able to reduce the list! Hopefully people still understand that they can be passed.